// vue-chart-3/dist/hooks.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.useScatterChart = exports.useBubbleChart = exports.useRadarChart = exports.usePolarAreaChart = exports.usePieChart = exports.useLineChart = exports.useBarChart = exports.useDoughnutChart = void 0;
const vue_demi_1 = require("vue-demi");
const defineChartHook = (chartType) => {
    return (params) => {
        const reactiveProps = vue_demi_1.computed(() => ({
            ...params,
            ref: `${chartType}ChartRef`,
            chartData: vue_demi_1.unref(params.chartData),
            options: vue_demi_1.unref(params.options),
        }));
        const chartProps = vue_demi_1.reactive({
            [`${chartType}ChartProps`]: reactiveProps,
        });
        return {
            ...vue_demi_1.toRefs(chartProps),
            [`${chartType}ChartRef`]: vue_demi_1.ref(),
        };
    };
};
exports.useDoughnutChart = defineChartHook('doughnut');
exports.useBarChart = defineChartHook('bar');
exports.useLineChart = defineChartHook('line');
exports.usePieChart = defineChartHook('pie');
exports.usePolarAreaChart = defineChartHook('polarArea');
exports.useRadarChart = defineChartHook('radar');
exports.useBubbleChart = defineChartHook('bubble');
exports.useScatterChart = defineChartHook('scatter');
